packages/application/src/connectionlost.ts

```diff
@@ -1,11 +1,14 @@
 // Copyright (c) Jupyter Development Team.
 // Distributed under the terms of the Modified BSD License.
 
-import { showErrorMessage } from '@jupyterlab/apputils';
+import { Dialog, showDialog } from '@jupyterlab/apputils';
 import { ServerConnection, ServiceManager } from '@jupyterlab/services';
 import { ITranslator, nullTranslator } from '@jupyterlab/translation';
 import { IConnectionLost } from './tokens';
 
+let displayConnectionLost = true;
+const connectionLostCachePromise: { [key: string]: Promise<any> } = {};
```

Comment:
New code should not use `any` except for tests. Any disables type checking and in past lead to numerous user-facing bugs. This will be enforced with https://github.com/jupyterlab/jupyterlab/pull/16219